RevenueCatUI/Templates/Components/Button/ButtonComponentViewModel.swift

```diff
@@ -20,8 +20,28 @@ import RevenueCat
 
 @available(iOS 15.0, macOS 12.0, tvOS 15.0, watchOS 8.0, *)
 public class ButtonComponentViewModel {
+    
+    /// A mirror of ButtonComponent.Action, with the sole purpose of being able to change the type of urlLid parameters
+    /// of some Destination's to actual URLs.
+    internal enum Action {
+        case restorePurchases
+        case navigateTo(destination: Destination)
+        case navigateBack
+    }
+
+    /// A mirror of ButtonComponent.Destination, with any urlLid parameters changed to be of the actual URL type. This
+    /// allows us to verify the URLs exist and are properly formatted, before making them available to the view layer.
+    /// This way the view layer doesn't need to handle this error scenario.
+    internal enum Destination {
+        case customerCenter
+        case URL(url: URL, method: PaywallComponent.ButtonComponent.URLMethod)
+        case privacyPolicy(url: URL, method: PaywallComponent.ButtonComponent.URLMethod)
+        case terms(url: URL, method: PaywallComponent.ButtonComponent.URLMethod)
+    }
```

Comment:
The goal here is for the `ButtonComponentView` to not have to worry about any "invalid config" scenarios. 

I also considered saving the `URL` in an optional field, which would always be non-nil in case we had a URL-like `Destination`. However, that would require the `ButtonComponentView` to trust the `ButtonComponentViewModel` on this (e.g. by force unwrapping). We would then need to write tests to ensure this "contract" never breaks through any refactorings. 

With the current approach, the `ButtonComponentView` reads a non-optional `URL` field, and thus it doesn't need to trust anyone (except the compiler), nor do we need to write a test for this case. 

Let me know if there's a better way of course!

Reply:
This looks like how I would have done it! 💪

Reply:
A compliment of the highest regard 🙇